src/ngx_http_set_hashed_upstream.c

```diff
@@ -5,18 +5,27 @@
 
 #include "ngx_http_set_hashed_upstream.h"
 
+#define HASH_VNODES           160
 
 ngx_uint_t
 ngx_http_set_misc_apply_distribution(ngx_log_t *log, ngx_uint_t hash,
-    ndk_upstream_list_t *ul, ngx_http_set_misc_distribution_t type)
+    ngx_http_set_hashed_upstream_conf *hash_conf,
+    ngx_http_set_misc_distribution_t type)
 {
+
     switch (type) {
     case ngx_http_set_misc_distribution_modula:
-        return (uint32_t) hash % (uint32_t) ul->nelts;
+        return (uint32_t) hash % (uint32_t) hash_conf->ul->nelts;
+
+    case ngx_http_set_misc_consistent_hash:
+        return hash_conf->hash_nodes[
+                ngx_http_set_misc_bsearch(hash_conf->hash_nodes,
+                                          hash_conf->node_len, hash) %
+                (uint32_t) hash_conf->node_len].index;
```

Comment:
It's better to calculate a local variable first, like `idx = ngx_http_set_misc_bsearch(hash_conf->hash_nodes,  hash_conf->node_len, hash) % (uint32_t) hash_conf->node_len;`
I personally think it will be more readable.